src/service-api.ts

```diff
@@ -122,29 +177,52 @@ const plugin: FastifyPluginAsync<GraaspChatPluginOptions> = async (
       },
     );
 
-    fastify.post<{ Params: { itemId: string }; Body: Partial<ChatMessage> }>(
+    fastify.post<{
+      Params: { itemId: string };
+      Body: Partial<PartialNewChatMessage>;
+    }>(
       '/:itemId/chat',
       { schema: publishMessage },
       async ({ member, params: { itemId }, body, log }) => {
         const tasks = taskManager.createPublishMessageTaskSequence(
           member,
           itemId,
-          body,
+          body.body,
         );
-        return runner.runSingleSequence(tasks, log);
+
+        const { message, mentionedUsers } = (await runner.runSingleSequence(
+          tasks,
+          log,
+        )) as { message: ChatMessage; mentionedUsers: (Member | Error)[] };
+        if (mentionedUsers && mentionedUsers.length > 0) {
+          const { result: item } = tasks[0];
+          mentionedUsers.forEach((mentionedUser) => {
+            if (!(mentionedUser instanceof Error)) {
+              sendMentionNotificationEmail({
+                item,
+                member: mentionedUser,
+                log,
+              });
+            }
```

Comment:
Wouldn't it make sense to send an email on create mention post hook? or after the sql query? I would prefer the posthook.

Reply:
@pyphilia Yes i guess it makes more sense. I have made the required changes. Feedback is welcome.